pkg/resmap/resmap.go

```diff
@@ -37,6 +37,26 @@ import (
 // ResMap is a map from ResId to Resource.
 type ResMap map[resource.ResId]*resource.Resource
 
+// FindByGVKN find the matched ResIds by Group/Version/Kind and Name
+func (m ResMap) FindByGVKN(inputId resource.ResId) []resource.ResId {
+	var result []resource.ResId
+	for id := range m {
+		if id.Name() == inputId.Name() && id.Gvk().String() == inputId.Gvk().String() {
```

Comment:
```
if id.Equals(inputId) {
}
```
i.e. write a (tested) `Equals` method for ResId.  In the guts, i'd not use String.  Be more explicit, e.g.
```
return  id.Gvk().Group == other.Gvk().Group() && id.Gvk().Version == ...
```
too bad the GroupVersionKind struct doesn't have an equals method already.